### PR TITLE
Update docs of `SHARD_ROW_ID_BITS`

### PR DESCRIPTION
**Pull request by dveeden**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

1. The change in default value for [`tidb_enable_clustered_index`](https://docs.pingcap.com/tidb/stable/system-variables#tidb_enable_clustered_index-new-in-v50) wasn't propagated to all docs as now non-integer PK's will also use a clustered index by default
2. A table without a primary key is not a good example.
3. Add links between related docs

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.0 (TiDB 7.0 versions)
- [x] v6.6 (TiDB 6.6 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/12904
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: shard-row-id-bits.md [hunk 1]
```diff
@@ -9,15 +9,27 @@ This document introduces the `SHARD_ROW_ID_BITS` table attribute, which is used
 
 ## Concept
 
-For the tables with a non-integer primary key or no primary key, TiDB uses an implicit auto-increment row ID. When a large number of `INSERT` operations are performed, the data is written into a single Region, causing a write hot spot.
+For the tables with a non-clustered primary key, no primary key or a multi-column primary key TiDB uses an implicit auto-increment row ID. When a large number of `INSERT` operations are performed, the data is written into a single Region, causing a write hot spot.
```
Comment (tangenta) [inline shard-row-id-bits.md:12]:
```suggestion
For the tables with a non-clustered primary key or no primary key, TiDB uses an implicit auto-increment row ID. When a large number of `INSERT` operations are performed, the data is written into a single Region, causing a write hot spot.
```

Multi-column primary key can be clustered, which does not have an implicit row ID:

```sql
CREATE TABLE t (
    id INT,
    c1 INT,
    PRIMARY KEY (id, c1) CLUSTERED
) SHARD_ROW_ID_BITS = 4;

ERROR 8200 (HY000): Unsupported shard_row_id_bits for table with primary key as row id
```

File: troubleshoot-hot-spot-issues.md [hunk 2]
```diff
@@ -87,9 +87,9 @@ Hover over the bright block, you can see what table or index has a heavy load. F
 
 ## Use `SHARD_ROW_ID_BITS` to process hotspots
 
-For a non-integer primary key or a table without a primary key or a joint primary key, TiDB uses an implicit auto-increment RowID. When a large number of `INSERT` operations exist, the data is written into a single Region, resulting in a write hotspot.
+For a non-clustered primary key or a table without a primary key or a multi-column primary key, TiDB uses an implicit auto-increment RowID. When a large number of `INSERT` operations exist, the data is written into a single Region, resulting in a write hotspot.
```
Comment (tangenta) [inline troubleshoot-hot-spot-issues.md:90]:
```suggestion
For a non-clustered primary key or a table without a primary key, TiDB uses an implicit auto-increment RowID. When a large number of `INSERT` operations exist, the data is written into a single Region, resulting in a write hotspot.
```

File: shard-row-id-bits.md [hunk 3]
```diff
@@ -9,15 +9,27 @@ This document introduces the `SHARD_ROW_ID_BITS` table attribute, which is used
...
 
 To mitigate the hot spot issue, you can configure `SHARD_ROW_ID_BITS`. The row IDs are scattered and the data are written into multiple different Regions.
 
 - `SHARD_ROW_ID_BITS = 4` indicates 16 shards
 - `SHARD_ROW_ID_BITS = 6` indicates 64 shards
 - `SHARD_ROW_ID_BITS = 0` indicates the default 1 shard
 
+See also [the Troubleshoot Hotspot Issues guide](/troubleshoot-hot-spot-issues.md#use-shard_row_id_bits-to-process-hotspots) for a guide about how to use this.
```
Comment (ran-huang) [inline shard-row-id-bits.md:20]:
```suggestion
For details on the usage, see [the Troubleshoot Hotspot Issues guide](/troubleshoot-hot-spot-issues.md#use-shard_row_id_bits-to-process-hotspots).

```

File: shard-row-id-bits.md [hunk 4]
```diff
@@ -9,15 +9,27 @@ This document introduces the `SHARD_ROW_ID_BITS` table attribute, which is used
...
 ## Examples
 
-- `CREATE TABLE`: `CREATE TABLE t (c int) SHARD_ROW_ID_BITS = 4;`
-- `ALTER TABLE`: `ALTER TABLE t SHARD_ROW_ID_BITS = 4;`
+{{< copyable "sql" >}}
+
```
Comment (Oreoxmt) [inline shard-row-id-bits.md:24]:
```suggestion
```
This copy syntax is no longer needed for the current docs website.

File: shard-row-id-bits.md [hunk 5]
```diff
@@ -9,15 +9,27 @@ This document introduces the `SHARD_ROW_ID_BITS` table attribute, which is used
...
+```sql
+CREATE TABLE t (
+    id INT PRIMARY KEY NONCLUSTERED
+) SHARD_ROW_ID_BITS = 4;
+```
+
+{{< copyable "sql" >}}
+
```
Comment (Oreoxmt) [inline shard-row-id-bits.md:32]:
```suggestion
```

**Review by tangenta — APPROVED** [review]

**Review by ran-huang — APPROVED** [review]

**Review by Oreoxmt — APPROVED** [review]
LGTM

**Comment by WalterWj** [comment]
LGTM

**Review by WalterWj — APPROVED** [review]

**Pull request merged**